### PR TITLE
Add export image functionality

### PR DESCRIPTION
**Pull request by bartvde**
SDK-511

### CODE REVIEW

File: examples/export-image/app.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  map: SdkMapReducer,
+  print: SdkPrintReducer,
+}), window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
+   applyMiddleware(thunkMiddleware));
+
+function main() {
+  const url = 'https://raw.githubusercontent.com/boundlessgeo/ol-mapbox-style/master/example/data/wms.json';
+  store.dispatch(mapActions.setContext({ url }));
+
+  const exportMapImage = (blob) => {
```
Comment (wnordmann) [inline examples/export-image/app.jsx]: A comment here about how this is the function called by the onExportImage prop would be useful

File: examples/export-image/app.jsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    saveAs(blob, 'map.png');
+    store.dispatch(printActions.receiveMapImage());
+  };
+
+  // place the map on the page.
+  ReactDOM.render(
```
Comment (wnordmann) [inline examples/export-image/app.jsx]: I think the export would be more interesting if there some some data included, even the random points would demo that layers are included in the export

File: src/components/map.jsx [hunk 3]
```diff
@@ -278,6 +278,11 @@ export class Map extends React.Component {
       this.updateInteraction(nextProps.drawing);
     }
 
+    if (nextProps.print && nextProps.print.exportImage) {
```
Comment (wnordmann) [inline src/components/map.jsx]: a comment here about using the canvas api would be helpful

File: examples/export-image/app.jsx (hunk 1 above)
Comment (bartvde) [inline examples/export-image/app.jsx]: good point, will add

File: examples/export-image/app.jsx (hunk 2 above)
Comment (bartvde) [inline examples/export-image/app.jsx]: there is the WMS USA states overlay, but it's currently not visible due to the center issue that duck just fixed, so should resolve itself

File: src/components/map.jsx (hunk 3 above)
Comment (bartvde) [inline src/components/map.jsx]: will add

File: examples/export-image/app.jsx (hunk 2 above)
Comment (bartvde) [inline examples/export-image/app.jsx]: I've rebased this branch so the center is now ok and shows the USA states layer for printing

**Comment by bartvde** [comment]
@wnordmann I've addressed your review with: https://github.com/boundlessgeo/sdk/pull/580/commits/82fb8d3e8458a4f6b43ae58cb9677a2819548313

@duck I will await your review

**Comment by bartvde** [comment]
@duck look at the example, once the image is received the toggle is flipped. So pressing the button again would work and result in a new image.

**Comment by bartvde** [comment]
@duck I didn't really like that the example code is responsible for flipping the switch though, but couldn't think of anything better, maybe you have other ideas for that

**Comment by bartvde** [comment]
@duck I just rebased this, let me know if you need any changes after I replied to your comment

**Comment by theduckylittle** [comment]
@bartvde  -- note my handle on here is @theduckylittle ... so someone else is probably really curious as to why they're getting notifications.

**Comment by bartvde** [comment]
ha, good catch ;-)

**Pull request merged**